R/derive_vars_trt_start.R

```diff
@@ -0,0 +1,152 @@
+#' Derive Datetime of Any Treatment Start Date
+#'
+#' Derives datetime of first exposure to treatment e.g. (`TRTSDTM`)
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by the `subject_keys` parameter are expected.
+#'
+#' @param dataset_ex `ex` dataset
+#'
+#'   The variables `EXSTDTC`, `EXSEQ`, and those specified by the `filter_ex`
+#'   `dtc` and `subject_keys`parameters are expected.
+#'
+#' @param filter_ex Filter condition for the ex dataset
+#'
+#'   Only observations of the ex dataset which fulfill the specified condition
+#'   are considered for the treatment start date.
+#'
+#'   Default: NULL
+#'
+#'   Permitted Values: logical expression
+#'
+#' @param subject_keys Variables to uniquely identify a subject
+#'
+#' A list of quosures where the expressions are symbols as returned by
+#' `vars()` is expected.
+#'
+#' @param new_var name of treatment start date e.g. `TRTSDTM`
+#'
+#' @param dtc name of variable from which start date is derived e.g. `EXSTDTC`
+#'
+#' @inheritParams impute_dtc
+#'
+#' @details For each group (with respect to the variables specified for the
+#'   `by_vars` parameter) the first observation (with respect to the order
+#'   specified for the `order` parameter) is included in the output dataset.
+#'
+#' @author Stefan Bundfuss, Teckla Akinyi
+#'
+#' @return The input dataset with start date variable added e.g. `TRTSDTM`
+#'
+#' @export
+#'
+#' @keywords adsl timing derivation
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#' data("ex")
+#' data("dm")
+#'
+#'dm %>%
+#'  derive_vars_trt_start(dataset_ex = ex,
+#'                        new_var=TRTSDTM,
+#'                        dtc=EXSTDTC,
+#'                        date_imputation = "first",
+#'                        time_imputation="first",
+#'                        min_dates = NULL,
+#'                        max_dates = NULL,
+#'                        order = vars(EXSTDTC,EXSEQ),
+#'                        subject_keys=vars(STUDYID,USUBJID),
+#'                        flag_imputation = "AUTO",
+#'                        filter_ex = EXDOSE>0 ,
+#'                        ord_filter="first") %>%
+#'  select(USUBJID, TRTSDTM)
+derive_vars_trt_start <- function(dataset,
+                                  dataset_ex,
+                                  filter_ex = NULL,
+                                  subject_keys,
+                                  new_var,
+                                  dtc,
+                                  date_imputation,
+                                  time_imputation,
+                                  min_dates = NULL,
+                                  max_dates = NULL,
+                                  flag_imputation = "AUTO",
```

Comment:
Could you please move `flag_imputation` just after `time_imputation` for consistency with other functions?